### ISSUE TITLE
TSQL (also ANSI?): Parse error when multiple columns specified in "ALTER TABLE"

### ISSUE THREAD

**Issue by SwiftyNZ** [sqlfluff/sqlfluff#3913]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Receiving unparsable section error when trying to alter a table and add multiple columns

### Expected Behaviour

Successfully passes linting

### Observed Behaviour

SQLFluff will report unparsable sections on drop column statements

### How to reproduce

Create file test.sql
```sql
ALTER TABLE my_table
ADD my_col_1 INT
  , my_col_2 INT
```
Run SQLFluff lint command
```
sqlfluff lint --dialect=tsql test.sql
```

Error message shown below
```
== [test.sql] FAIL
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'ALTER
                       | TABLE my_table\nADD my_col_1\n\t, my_...'
WARNING: Parsing errors found and dialect is set to 'tsql'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### Dialect

tsql

### Version

```
> sqlfluff --version
sqlfluff, version 1.3.2
```

### Configuration

[sqlfluff]
dialect = tsql
sql_file_exts = .sql
large_file_skip_char_limit = 20000
exclude_rules = L031, L034

[sqlfluff:rules]
comma_style = leading
tab_space_size = 4
max_line_length = 120
indent_unit = tab
single_table_references = unqualified

[sqlfluff:indentation]
indented_joins = False
indented_ctes = False

[sqlfluff:rules:L010]
capitalisation_policy = upper

[sqlfluff:rules:L011]
aliasing = explicit

[sqlfluff:rules:L014]
extended_capitalisation_policy = lower
unquoted_identifiers_policy = all

[sqlfluff:rules:L030]
extended_capitalisation_policy = upper

[sqlfluff:rules:L063]
extended_capitalisation_policy = upper

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by adam-tokarski** [comment]
This is issue not only for `t-sql`. The thing is that for `tsql` it prints the error message while for `ansi`, for example, it finishes linting without any issue, but most likely it treats the second column definition as something else:
```
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        alter_table_statement:
[L:  1, P:  1]      |            keyword:                                          'ALTER'
[L:  1, P:  6]      |            whitespace:                                       ' '
[L:  1, P:  7]      |            keyword:                                          'TABLE'
[L:  1, P: 12]      |            whitespace:                                       ' '
[L:  1, P: 13]      |            table_reference:
[L:  1, P: 13]      |                naked_identifier:                             'my_table'
[L:  1, P: 21]      |            whitespace:                                       ' '
[L:  1, P: 22]      |            keyword:                                          'ADD'
[L:  1, P: 25]      |            whitespace:                                       ' '
[L:  1, P: 26]      |            column_definition:
[L:  1, P: 26]      |                naked_identifier:                             'my_col_1'
[L:  1, P: 34]      |                whitespace:                                   ' '
[L:  1, P: 35]      |                data_type:
[L:  1, P: 35]      |                    data_type_identifier:                     'INT'
[L:  1, P: 38]      |            comma:                                            ','
[L:  1, P: 39]      |            whitespace:                                       ' '
[L:  1, P: 40]      |            parameter:                                        'my_col_2'
[L:  1, P: 48]      |            whitespace:                                       ' '
[L:  1, P: 49]      |            naked_identifier:                                 'int'
[L:  1, P: 52]      |    statement_terminator:                                     ';'
[L:  1, P: 53]      |    newline:                                                  '\n'
[L:  2, P:  1]      |    [META] end_of_file:
```

**Comment by AidanHarveyNelson** [comment]
@adam-tokarski Do you have any documentation on whether multiple columns with ADD is supported for ANSI.

I can see that most DBMS have support for it but they seem to have their own flavor (With or without brackets for example)
